### PR TITLE
feat: add Grumpkin elliptic curve (native 2-cycle with BN254)

### PR DESCRIPTION
**Pull request by yelhousni**
# Description

This PR implements arithmetic over the Grumpkin elliptic curve as native operations in a BN254 SNARK. This is similar to `sw_bls12377` and `sw_bls24315` packages, except there is no (efficient) pairing on Grumpkin. 

[Grumpkin](https://aztecprotocol.github.io/aztec-connect/primitives.html/) forms a 2-cycle with BN254 and the combination is useful for many protocols.

TODO:
- [ ] implement endomorphism-based optimizations.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

Arithmetic tests against gnark-crypto are implemented + edge cases.

# How has this been benchmarked?

|   | r1cs | scs |
|--|-----|----|
| scalar mul | 3,634 | 8,797 |
| joint scalar mul | 3,563 | 8,954 |

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I did not modify files generated from templates
- [x] `golangci-lint` does not output errors locally
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: std/algebra/native/sw_grumpkin/doc.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Package sw_grumpkin implements the arithmetics on ithe Grumpkin curve as a
+// SNARK circuit over BN254. These two curves form a 2-cycle so the operations
+// use native field arithmetic.
+//
+// References:
+// https://aztecprotocol.github.io/aztec-connect/primitives.html/
```
Comment (Saw-mon-and-Natalie) [inline std/algebra/native/sw_grumpkin/doc.go:6]:
```suggestion
// https://aztecprotocol.github.io/aztec-connect/primitives.html
```

File: std/algebra/native/sw_grumpkin/hints.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		s1 = api.Add(s1, api.Mul(sd[0].Limbs[i], b))
+		s2 = api.Add(s2, api.Mul(sd[1].Limbs[i], b))
+		b.Lsh(b, 64)
+	}
+	return s1, s2
+}
+
+func decompose(mod *big.Int, inputs, outputs []*big.Int) error {
+	if len(inputs) != 1 && len(outputs) != 4 {
+		return errors.New("input/output length mismatch")
```
[the remark this answers is not in the document]
Comment (ivokub) [reply std/algebra/native/sw_grumpkin/hints.go:86]: Fixed
